appservice/src/createAppService/SetWizardContextPropertyStep.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from 'vscode-azureextensionui';
+import { IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+/**
+ * This step is used to alter the WizardContext's properties without having to recreate the wizard or
+ * stop the prompts.  This is neccessary when the value of a property relies on previously prompted
+ * steps i.e. the resource group's name relies on the app location so a location must be selected first
+ */
+export class SetWizardContextPropertyStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
```

Comment:
Yeah I know you added comments to this class to make it less confusing, but I still don't get why this class is necessary. Maybe you can explain it to me offline and we can brainstorm ways to possibly get rid of it